ext-dynamic-commitments.md

```diff
@@ -0,0 +1,1087 @@
+convert existing channels into Taproot channels.
+
+# Specification
+
+There are two phases to this channel upgrade process: proposal, and execution.
+During the proposal phase the only goal is to agree on a set of updates to the
+current channel state machine. Assuming an agreement can be reached, we will
+proceed to the execution phase. During the execution phase, we apply the updates
+to the channel state machine, exchanging the necessary information to be able to
+apply those updates.
+
+## Proposal Phase
+
+As a prerequisite to the proposal phase of a Dynamic Commitment negotiation, the
+channel must be in a [quiesced](https://github.com/lightning/bolts/pull/869)
+state.
+
+### Node Roles
+
+In every dynamic commitment negotiation, there are two roles: the `initiator`
+and the `responder`. It is necessary for both nodes to agree on which node is
+the `initiator` and which node is the `responder`. This is important because if
+the dynamic commitment negotiation results in a re-anchoring step (described
+later), it is the `initiator` that is responsible for paying the fees for the
+kickoff transaction. The `initiator` is determined by who has the `initiator`
+role established by the quiescence process.
+
+### Negotiation TLVs
+
+The following TLVs are used throughout the negotiation phase of the protocol
+and are common to all messages in the negotiation phase.
+
+#### dust_limit_satoshis
+
+- type: 0
+  data:
+    * [`u64`:`dust_limit_satoshis`]
+
+#### max_htlc_value_in_flight_msat
+
+- type: 1
+  data:
+    * [`u64`:`senders_max_htlc_value_in_flight_msat`]
+
+#### channel_reserve_satoshis
+
+- type: 2
+  data:
+    * [`u64`:`recipients_channel_reserve_satoshis`]
+
+#### to_self_delay
+
+- type: 3
+  data:
+    * [`u16`:`recipients_to_self_delay`]
+
+#### max_accepted_htlcs
+
+- type: 4
+  data:
+    * [`u16`:`senders_max_accepted_htlcs`]
+
+#### funding_pubkey
+
+- type: 5
+  data:
+    * [`point`:`senders_funding_pubkey`]
+
+#### channel_type
+
+- type: 6
+  data:
+    * [`...*byte`:`channel_type`]
+
+#### kickoff_feerate_per_kw
+
+- type: 7
+  data:
+    * [`u32`:`kickoff_feerate_per_kw`]
+
+
+### Proposal Messages
+
+Three new messages are introduced that are common to all dynamic commitment
+flows. They let each channel party propose which channel parameters they wish to
+change as well as accept or reject the proposal made by their counterparty.
+
+#### `dyn_propose`
+
+This message is sent to negotiate the parameters of a dynamic commitment
+upgrade. The overall protocol flow is depicted below. This message is always
+sent by the `initiator`.
+
+        +-------+                               +-------+
+        |       |--(1)------ dyn_propose ------>|       |
+        |   A   |                               |   B   |
+        |       |<-(4)---{dyn_ack|dyn_reject}---|       |
+        +-------+                               +-------+
+
+1. type: 111 (`dyn_propose`)
```

Comment:
Feature bit + even message number?

Reply:
Definitely the right move. This will be addressed. I haven't given as much attention to the specifics of the type/message number selection as I've been focusing on the overall shape of the design, but it's definitely important to make sure this matches the expectations of the rest of the protocol.